src/oref_alert_indexer.py

```diff
@@ -0,0 +1,121 @@
+import pandas as pd
+import matplotlib.pyplot as plt
+import warnings
+warnings.filterwarnings('ignore')
+import requests
+import json
+from datetime import datetime, timedelta
+import time 
+from errors import OrefAPIException, RateLimitException, SqlDatabaseException
+import os
+from dotenv import load_dotenv, dotenv_values
+import logging
+from sql_database import SqlOrefDatabase
+
+load_dotenv()
+URL = os.getenv('ALERTS_URL')
+ALERTS_DATA_FILE = os.getenv('ALERTS_DATA_FILE')
+DAYS_INTERVAL = 2
+SLEEP_TIME = 1
+DELAYTIME = 5
+RETRIESNUM = 3
+
+# Retrieve alerts from oref API by given range of time, and build database
+class OrefAlertsIndexer:
+    # Decorator function for get_wikipedia_page function
+    def retry_on_rate_limit_exception(func):
```

Comment:
move this decorator to a different file, call it `utils.py ` and import that decorator from there

Reply:
get rid of the const and use params:
```
def retry_on_rate_limit_exception(func, retries: int, delay: float = 1.0):
```